File: Server/src/main/java/bgu/spl171/net/impl/packets/ERRORPacket.java
package bgu.spl171.net.impl.packets;

/**
 * Error Packet
 */
public class ERRORPacket extends Packet {

    private short errorCode;
    private String errMsg;

    /**
     * Constructor
     * @param errorCode The error code
     * @param errMsg The error message
     */
    public ERRORPacket(short errorCode, String errMsg) {
        opCode = ERROR;
        this.errorCode = errorCode;
        this.errMsg = errMsg;
    }

    /**
     * Getter for the error code
     * @return the error code
     */
    public short getErrorCode() {
        return errorCode;
    }

    /**
     * Getter for the error message
     * @return the error message
     */
    public String getErrMsg() {
        return errMsg;
    }
}

File: Client/src/ERRORPacket.cpp
#include "ERRORPacket.h"

ERRORPacket::ERRORPacket(short errorCode, const std::string & errorMessage) : Packet(ERROR), errorCode_(errorCode), errorMessage_(errorMessage) {}

short ERRORPacket::getErrorCode() const {
	return errorCode_;
}

std::string ERRORPacket::getErrorMessage() const {
	return errorMessage_;
}
